admin/app/components/solidus_admin/ui/tab/component.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Tab::Component < SolidusAdmin::BaseComponent
+  SIZES = {
+    s: %w[h-7 px-1.5 body-small-bold],
+    m: %w[h-9 px-3 body-small-bold],
+    l: %w[h-12 px-4 body-text-bold],
+  }
+
+  TAG_NAMES = {
+    a: :a,
+    button: :button,
+  }
+
+  def initialize(text:, size: :m, tag: :a, disabled: false, active: false, **attributes)
```

Comment:
I overlooked that in the first round, but aren't `disabled` & `active` mutually exclusive? Do you think we could reflect that better by having a `state` "enum" parameter as input with `:default`, `:active` or `:disabled` as possible values?

Reply:
I started out with a state, but then moved to independent args, mainly because I noticed that for a button `disabled` would be an attribute and I would expect to be able to pass `disabled: true` like in other components.

I'm clearly conflicted in the decision, but in the current state `disabled` takes precedence over `active` which seems correct.

Thoughts?

Reply:
I didn't think about that, but maybe we should not see a tab as a button and couple our UI element's behavior with the HTML element. No strong feelings here, but I think that we should change the interface of the component only because one of its elements might expect a certain pattern.

Reply:
Yeah, I also think that we should build the interface that makes sense for this given component :+1:

Reply:
I'm not sure what you are suggesting (sorry, it's Friday! 😅).

I think both options make sense, and passing `disabled:` will be common to every other `input` element.

If you want to see the component in its usage context here it is:

```erb
   <div class="<%= toolbar_classes %>" data-<%= stimulus_id %>-target="scopesToolbar">
     <%= render component('ui/tab').new(text: "All", active: true, scheme: :secondary, href: "") %>
   </div>
```

https://github.com/solidusio/solidus/pull/5180/files#diff-b52bed4cf337144516f6db24019235696a9e901ab2cd0f96ef06fbd06560e4a2R12-R14

Reply:
I see the tab more as a navigational element than a for object, so I think having the `a` element is enough to cover all the cases. But until we see it in action we don't know yet and it might be also useful as a `button` sometimes. Up to you, I'm also ok with the current version.

Reply:
I understand your point about the `button` tag. What I find annoying is something like:

```ruby
component('ui/tab').new(text: "All", active: true, disabled: true)
```

The method signature does not define it, and I need to read the implementation to understand the behavior. That's because, in nature, we're dealing with an enum type.

@elia, do we have somewhere in the design files where a tab acts like a button? That could help to make the decision. If we have one, or we foresee we'll have one, would it be better to create another button component? In the end, matching styles should be seen as a coincidence, but a I think a component should wrap the UI or domain nature of the elements, not its style. Thoughts?

Reply:
@waiting-for-dev 

My basic thought about this is that I don't consider any of this final, and I would spend as little as possible guessing vs. sourcing the best API we can from actual usage in the application. Right now we only have one use case for this component which is inside `ui/table` for the scopes. 

The only reason I went for separate fields is that is reasonable to expect `disabled` to be an independent attribute which would also be reflected in the DOM if used with a button element.

Since I don't know yet what kind of interaction we'll have between that "scopes bar" and filters/searches/sorting on the table I can't predict if it's gonna be one or the other. So I quickly did what kind of made sense delaying a refinement of the API to when we'll have more information. I'm certain that using a button element is possible as the filter/search/sorting system _can_ end up using an underlying form, but again, I'm not sure yet.

I also already pointed out that when both `active` and `disabled` are set te component appears as disabled. To me the current situation is good enough as a basis for further refinement down the road and I would move along as fast as possible.

The other option, also popular among teams building components, is to be more strict and not extract anything to the "UI" library until it's used at least in two different components. That would include `ui/table` become `products/index/table` and so on, until we have another usage place. I would be ok with that as long as we're able to swiftly rename components.

_Sorry for the wall of text, I how this gives you a better picture of how I came up with this code and why I'm resisting a conversation on which API would be "better". 🙏_

Reply:
Fair enough, @elia. Let's go ahead and move forward. In my experience, these refinements left for later are not often reviewed later on. Because of that, I always prefer to build an API that makes sense for the current code version, as it's always easier to revisit when driven by an actual need to extend. But you're more familiar with these components as part of your work with batch actions, so let's merge this!

Reply:
Thanks @waiting-for-dev, really appreciate ❤️